screenpy/actions/eventually.py

```diff
@@ -158,6 +160,15 @@ def __init__(self, performable: Performable):
         self.performable = performable
         self.performable_to_log = get_additive_description(self.performable)
         self.caught_error = None
-        self.unique_errors: Dict[Exception, None] = {}
+        self.unique_errors: List[BaseException] = []
         self.timeout = settings.TIMEOUT
         self.poll = settings.POLLING
+
+
+def same_exception(a: BaseException, b: BaseException) -> bool:
+    """compare exceptions to see if they match"""
+    return (
+        isinstance(a, type(b))
+        and (str(a) == str(b))
+        and (format_tb(a.__traceback__) == format_tb(b.__traceback__))
+    )
```

Comment:
Some things i'd like to see for this function:
* The name, maybe `isequal_exception`? Kinda like `isinstance`?
* the parameters, we can do better than `a` and `b`. Maybe even `first_exc` and `second_exc`?
* The docstring, i think it can be a little more explicit about what we're doin' here.

Reply:
I also have mixed feelings about this function hanging out in `Eventually`, but i can't figure out what exactly makes me feel like i want to avoid that. It being on the outside means we can use it elsewhere if we need to, which is nice, but is this the right place for it?

Reply:
I'm glad you said something.  I almost put it into a module called "common" or some such.  But YAGNI prevented me.  If we ever need the function elsewhere, we can move it then.